eth/gasprice/gasprice.go

```diff
@@ -167,19 +409,89 @@ func (gpo *Oracle) getBlockPrices(ctx context.Context, signer types.Signer, bloc
 		return
 	}
 
+	blockTxs := block.Transactions()
+	txs := new(transactionsByGasPrice)
+	txs.txs = make([]*types.Transaction, len(blockTxs))
+	copy(txs.txs, blockTxs)
+	txs.baseFee = block.BaseFee()
+	sort.Sort(txs)
+
+	for _, tx := range txs.txs {
+		sender, err := types.Sender(signer, tx)
+		if err == nil && sender != block.Coinbase() {
+			price := tx.GasPrice()
+			if price == nil {
+				price = new(big.Int).Add(block.BaseFee(), tx.GasPremium())
+				if price.Cmp(tx.FeeCap()) > 0 {
+					price.Set(tx.FeeCap())
+				}
+			}
+			ch <- getBlockPricesResult{price, nil}
+			return
+		}
+	}
+	ch <- getBlockPricesResult{nil, nil}
+}
+
+// getBlockPremiums calculates the lowest transaction gas premium in a given block
+// and sends it to the result channel. If the block is empty, price is nil.
+func (gpo *Oracle) getBlockPremiums(ctx context.Context, signer types.Signer, blockNum uint64, ch chan getBlockPremiumsResult) {
+	block, err := gpo.backend.BlockByNumber(ctx, rpc.BlockNumber(blockNum))
+	if block == nil {
+		ch <- getBlockPremiumsResult{nil, err}
+		return
+	}
+
+	blockTxs := block.Transactions()
+	txs := new(transactionsByGasPremium)
+	txs.txs = make([]*types.Transaction, len(blockTxs))
+	copy(txs.txs, blockTxs)
+	txs.baseFee = block.BaseFee()
+	sort.Sort(txs)
+
+	for _, tx := range txs.txs {
+		sender, err := types.Sender(signer, tx)
+		if err == nil && sender != block.Coinbase() {
+			premium := tx.GasPremium()
+			if premium == nil {
+				premium = new(big.Int).Sub(tx.GasPrice(), block.BaseFee())
+				if premium.Cmp(common.Big0) < 0 {
+					premium.Set(common.Big0)
+				}
+			}
+			ch <- getBlockPremiumsResult{premium, nil}
+			return
+		}
+	}
```

Comment:
```suggestion
		if err == nil && sender != block.Coinbase() {
		        continue
		}
		premium := tx.GasPremium()
		if premium == nil {
			premium = new(big.Int).Sub(tx.GasPrice(), block.BaseFee())
			if premium.Cmp(common.Big0) < 0 {
				premium.Set(common.Big0)
			}
		}
		ch <- getBlockPremiumsResult{premium, nil}
		return
	}
```
Same here

Reply:
Looks good to me, although the conditions need to be switched on the `continue` check.